gui/src/components/LayoutComponents/NavBar.tsx

```diff
@@ -1,18 +1,24 @@
 'use client';
 import './style.css';
-import { useEffect, useRef, useState } from 'react';
+import { useContext, useEffect, useState } from 'react';
 import Image from 'next/image';
 import imagePath from '@/app/imagePath';
 import { navbarOptions } from '@/app/constants/NavbarConstants';
 import { useRouter } from 'next/navigation';
 import useProjectDropdown from '@/hooks/useProjectDropdown';
 import CustomImage from '@/components/ImageComponents/CustomImage';
 import CustomDropdown from '@/components/CustomDropdown/CustomDropdown';
-import { logout } from '@/app/utils';
+import { getUsernameInitials, logout } from '@/app/utils';
 import CreateOrEditProjectBody from '@/components/HomeComponents/CreateOrEditProjectBody';
+import { UserContext } from '@/context/UserContext';
 
 export default function NavBar() {
-  const [username, setUsername] = useState<string | null>('Major Pandey Singh');
+  const userContext = useContext(UserContext);
+  if (!userContext) {
```

Comment:
move this to UserContext.tsx itself will avoid duplicate code and others forgetting to add it by mistake.